content/tutorials/spatial_wfs_services/index.Rmd

```diff
@@ -498,6 +504,96 @@ knitr::kable(df)
+#' Flemish region, an NA value is given for each of the properties.
+#'
+#' @param x_lam The numeric value of the X coordinate as Lambert 72.
+#' @param y_lam The numeric value of the X coordinate as Lambert 72.
+#' @param properties_of_interest A vector or properties, as a subset of these
+#' provided by the webservice. Default Bodemserie, Unibodemtype and Bodemtype.
+#'
+#' @return A data.frame with the properties as column headers.
+#'
+#' @importFrom httr GET
+#' @importFrom jsonlite fromJSON
+#'
+#' @examples
+#' soil_map_data_extraction(173995.67, 212093.44)
+#' soil_map_data_extraction(173995.67, 212093.44,
+#'                          properties_of_interest = c("Eenduidige_legende",
+#'                                                     "Textuurklasse"))
+soil_map_data_extraction <- function(x_lam,
+                                     y_lam,
+                                     properties_of_interest = c("Bodemserie",
+                                                                "Unibodemtype",
+                                                                "Bodemtype")) {
+  if (missing(x_lam) | missing(y_lam)) stop("x_lam and y_lam needed")
+  # dealing with point data inside a certain polygon of the soil map:
+  wfs_bodemtypes <- "https://www.dov.vlaanderen.be/geoserver/bodemkaart/bodemtypes/wfs?"
+  query = list(service = "WFS",
+               request = "GetFeature",
+               version = "1.1.0",
+               typeName = "bodemkaart:bodemtypes",
+               outputFormat = "json",
+               propertyname = as.character(paste(properties_of_interest,
+                                                 collapse = ",")),
+               CRS = "EPSG:31370",
+               CQL_FILTER = sprintf("INTERSECTS(geom,POINT(%s %s))",
+                                    x_lam, y_lam)) # INTERSECT OPERATOR
+
+  result <- httr::GET(wfs_bodemtypes, query = query)
+  parsed <- jsonlite::fromJSON(httr::content(result, "text"))
+  soil_info_df <- parsed$features
+  # if else toegevoegd om gevallen op te vangen waarbij een punt (net) buiten de kaart valt
+  if (is.null(soil_info_df$properties)) {
+    as.data.frame(
+      matrix(rep(NA, length(properties_of_interest)),
+             nrow = 1,
+             dimnames = list(NULL, properties_of_interest)))
+  } else {
+    soil_info_df$properties[,-1]
+  }
+
+}
+
+```
+
+Using this function to extract soil data features is a lot more user-friendly:
+
+```{r}
+soil_map_data_extraction(
+  x_lam = x_lam, 
+  y_lam = y_lam)
+
+soil_map_data_extraction(
+  x_lam = x_lam, 
+  y_lam = y_lam,
+  properties_of_interest = c("Eenduidige_legende", "Textuurklasse"))
+
+soil_map_data_extraction(
+  x_lam = x_lam, 
+  y_lam = y_lam,
+  properties_of_interest = properties_of_interest)
+
+# multiple point locations
+xy <- data.frame(id = c("loc1", "loc2"),
+                 x = c(173995.67, 180000),
+                 y = c(212093.4, 212000))
+
+xy %>%
```

Comment:
Why not have the function to handle a matrix of coordinates? Or a SpatialPointsDataFrame?

Reply:
Good idea. I can see if it is possible to send more than one point at the time via the wfs request. If this is not the case, maybe it's better to leave it up to the user to apply the function repeatedly as shown in the example?
BTW: the function is moved to here: https://github.com/inbo/inborutils/pull/94

Reply:
I decided not to do that. The tutorial will show how this can be easily done.